test/NITTA/Model/ProcessorUnits/Tests/DSL/Tests.hs

```diff
@@ -290,6 +293,28 @@ tests =
                             , Pull $ O $ S.fromList ["d^0#2"]
                             ]
                         ]
+            , unitTestCase "target system: manual synthesis, allocation works correctly" def $ do
+                setNetwork $
+                    Bus.defineNetwork "net1" ASync $ do
+                        Bus.addPrototype "fram{x}" FramIO
+                        Bus.addPrototype "accum" AccumIO
+                setBusType pInt
+                assignLua
+                    [__i|
+                        function sum(a)
+                            local d = a + 1
+                            sum(d)
+                        end
+                        sum(0)
+                        |]
+                doAllocation "net1" "accum"
+                doAllocation "net1" "fram{x}"
```

Comment:
Related to `string manipulation` and `add the type of allocated PU ` comments, I think Proxy type will be better, but we can return to it in other PR.

Reply:
Yeah, let's do this in separate PR after discussing #215